src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Completion/RazorCompletionEndpoint.cs

```diff
@@ -227,8 +223,14 @@ public Task<CompletionItem> Handle(CompletionItem completionItem, CancellationTo
             return Task.FromResult(completionItem);
         }
 
+        public void SetCapability(CompletionCapability capability)
+        {
+            _capability = capability;
+            _tagHelperCompletionService.SetCapability(_capability);
```

Comment:
Instead of this pattern i'd just have the implementation of the `TagHelperCompletionService` take in the [ILanguageServer](https://github.com/OmniSharp/csharp-language-server-protocol/blob/179d21d9fa838496640541fec8a216d292721d41/src/Server/ILanguageServer.cs#L7) and extract its `ClientSettings` which have the current [Capabilities](https://github.com/OmniSharp/csharp-language-server-protocol/blob/179d21d9fa838496640541fec8a216d292721d41/src/Protocol/Models/InitializeParams.cs).

That way you don't need to have a `SetCapability` method on the completion service and any other dependent types.